rpc/src/main/scala/org/bitcoins/rpc/client/RpcClient.scala

```diff
@@ -0,0 +1,1114 @@
+    signRawTransaction(transaction, Some(utxoDeps), Some(keys), None)
+
+  def signRawTransaction(
+      transaction: Transaction,
+      utxoDeps: Vector[RpcOpts.SignRawTransactionOutputParameter],
+      keys: Vector[ECPrivateKey],
+      sigHash: String): Future[SignRawTransactionResult] =
+    signRawTransaction(transaction, Some(utxoDeps), Some(keys), Some(sigHash))
+
+  def stop(): Future[String] = {
+    bitcoindCall[String]("stop")
+  }
+
+  def submitBlock(block: Block): Future[Unit] = {
+    bitcoindCall[Unit]("submitblock", List(JsString(block.hex)))
+  }
+
+  def uptime: Future[UInt32] = {
+    bitcoindCall[UInt32]("uptime")
+  }
+
+  def validateAddress(
+      address: BitcoinAddress): Future[ValidateAddressResult] = {
+    bitcoindCall[ValidateAddressResult]("validateaddress",
+                                        List(JsString(address.toString)))
+  }
+
+  def verifyChain(level: Int = 3, blocks: Int = 6): Future[Boolean] = {
+    bitcoindCall[Boolean]("verifychain",
+                          List(JsNumber(level), JsNumber(blocks)))
+  }
+
+  def verifyMessage(
+      address: P2PKHAddress,
+      signature: String,
+      message: String): Future[Boolean] = {
+    bitcoindCall[Boolean](
+      "verifymessage",
+      List(JsString(address.value), JsString(signature), JsString(message)))
+  }
+
+  def verifyTxOutProof(
+      proof: MerkleBlock): Future[Vector[DoubleSha256Digest]] = {
+    bitcoindCall[Vector[DoubleSha256Digest]]("verifytxoutproof",
+                                             List(JsString(proof.hex)))
+  }
+
+  def walletLock(): Future[Unit] = {
+    bitcoindCall[Unit]("walletlock")
+  }
+
+  def walletPassphrase(passphrase: String, seconds: Int): Future[Unit] = {
+    bitcoindCall[Unit]("walletpassphrase",
+                       List(JsString(passphrase), JsNumber(seconds)))
+  }
+
+  def walletPassphraseChange(
+      currentPassphrase: String,
+      newPassphrase: String): Future[Unit] = {
+    bitcoindCall[Unit](
+      "walletpassphrasechange",
+      List(JsString(currentPassphrase), JsString(newPassphrase)))
+  }
+
+  private def bitcoindCall[T](
+      command: String,
+      parameters: List[JsValue] = List.empty)(
+      implicit
+      reader: Reads[T]): Future[T] = {
+    val request = buildRequest(instance, command, JsArray(parameters))
+    val responseF = sendRequest(request)
+
+    val payloadF: Future[JsValue] = responseF.flatMap(getPayload)
+
+    payloadF.map { payload =>
+      parseResult((payload \ resultKey).validate[T], payload)
+    }
+  }
+
+  case class RpcError(code: Int, message: String)
+  implicit val rpcErrorReads: Reads[RpcError] = Json.reads[RpcError]
+
+  // Should both logging and throwing be happening?
+  private def parseResult[T](result: JsResult[T], json: JsValue): T = {
+    // First catch errors thrown by calls with Unit as the expected return type (which isn't handled by UnitReads)
+    if (result == JsSuccess(())) {
```

Comment:
Let's factor this out into it's own method. Call it `checkUnit()` or something? Add documentation to this method.